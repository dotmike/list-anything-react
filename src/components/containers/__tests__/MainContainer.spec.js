import configureStore from "redux-mock-store";

import ConnectedMainContainer, {
  UnconnectedMainContainer
} from "../MainContainer";

const setup = (propOverrides = {}) => {
  const props = {
    menuVisible: false,
    hideMenu: jest.fn(),
    search: jest.fn(),
    children: "Testando...",
    activeItem: "",
    subjects: [],
    ...propOverrides
  };

  return {
    props,
    connectedWrapperShallow: wrapperShallow(
      ConnectedMainContainer,
      props,
      true
    ),
    wrapperShallow: wrapperShallow(UnconnectedMainContainer, props, true)
  };
};

const testData = {};
testData.subjects = [{ _id: "1", description: "Test" }];
testData.activeItem = `/subject/${testData.subjects[0]._id}`;
testData.to = testData.activeItem;

describe("ConnectedMainContainer", () => {
  const mockStore = configureStore();
  const state = {};
  it("renders correctly", () => {
    const { connectedWrapperShallow: wrapper } = setup({
      store: mockStore(state)
    });
    expect(wrapper()).toMatchSnapshot();
  });
});

describe("UnconnectedMainContainer", () => {
  it("renders correctly when `subjects` is empty", () => {
    const { wrapperShallow: wrapper } = setup({ subjects: [] });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `subjects` is not empty", () => {
    const { wrapperShallow: wrapper } = setup({
      subjects: testData.subjects
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `menuVisible` is false", () => {
    const { wrapperShallow: wrapper } = setup({
      menuVisible: false
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `menuVisible` is true", () => {
    const { wrapperShallow: wrapper } = setup({
      menuVisible: true
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `activeItem` is set", () => {
    const { wrapperShallow: wrapper } = setup({
      subjects: testData.subjects,
      activeItem: testData.activeItem
    });
    expect(wrapper()).toMatchSnapshot();
  });

  it("calls `hideMenu` when clicking on a `Sidemenu Link`", () => {
    const { wrapperShallow: wrapper, props } = setup({
      subjects: testData.subjects
    });

    wrapper()
      .find({ to: testData.to })
      .simulate("click");
    expect(props.hideMenu).toHaveBeenCalled();
  });
});
