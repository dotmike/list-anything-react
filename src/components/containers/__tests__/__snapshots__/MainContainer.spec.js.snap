// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedMainContainer renders correctly 1`] = `
<MainContainer
  activeItem=""
  dispatch={[Function]}
  hideMenu={[Function]}
  menuVisible={false}
  search={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  subjects={Array []}
>
  Testando...
</MainContainer>
`;

exports[`UnconnectedMainContainer renders correctly when \`activeItem\` is set 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  id="main-container"
>
  <Sidebar
    animation="uncover"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  >
    <MenuItem
      className="hide-gt-767px"
    >
      <SearchInput
        id="seach-form-sidebar"
        onSearch={[Function]}
      />
    </MenuItem>
    <MenuItem
      active={true}
      as={[Function]}
      key="1"
      onClick={[Function]}
      to="/subject/1"
    >
      Test
    </MenuItem>
  </Sidebar>
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedMainContainer renders correctly when \`menuVisible\` is false 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  id="main-container"
>
  <Sidebar
    animation="uncover"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  >
    <MenuItem
      className="hide-gt-767px"
    >
      <SearchInput
        id="seach-form-sidebar"
        onSearch={[Function]}
      />
    </MenuItem>
  </Sidebar>
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedMainContainer renders correctly when \`menuVisible\` is true 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  id="main-container"
>
  <Sidebar
    animation="uncover"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={true}
    width="thin"
  >
    <MenuItem
      className="hide-gt-767px"
    >
      <SearchInput
        id="seach-form-sidebar"
        onSearch={[Function]}
      />
    </MenuItem>
  </Sidebar>
  <SidebarPusher
    dimmed={true}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedMainContainer renders correctly when \`subjects\` is empty 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  id="main-container"
>
  <Sidebar
    animation="uncover"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  >
    <MenuItem
      className="hide-gt-767px"
    >
      <SearchInput
        id="seach-form-sidebar"
        onSearch={[Function]}
      />
    </MenuItem>
  </Sidebar>
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;

exports[`UnconnectedMainContainer renders correctly when \`subjects\` is not empty 1`] = `
<SidebarPushable
  as={[Function]}
  attached="bottom"
  id="main-container"
>
  <Sidebar
    animation="uncover"
    as={[Function]}
    direction="left"
    icon="labeled"
    inverted={true}
    vertical={true}
    visible={false}
    width="thin"
  >
    <MenuItem
      className="hide-gt-767px"
    >
      <SearchInput
        id="seach-form-sidebar"
        onSearch={[Function]}
      />
    </MenuItem>
    <MenuItem
      active={false}
      as={[Function]}
      key="1"
      onClick={[Function]}
      to="/subject/1"
    >
      Test
    </MenuItem>
  </Sidebar>
  <SidebarPusher
    dimmed={false}
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    Testando...
  </SidebarPusher>
</SidebarPushable>
`;
